eventbridge-pipes-msk-iam-auth-to-lambda/msk-cluster.yaml

```diff
@@ -0,0 +1,184 @@
+      - Description: IAM Auth access control
+        FromPort: 9098
+        ToPort: 9098
+        IpProtocol: tcp
+        SourceSecurityGroupId: !GetAtt KafkaClientInstanceSecurityGroup.GroupId
+      - Description: Bootstrap servers TLS
+        FromPort: 9094
+        ToPort: 9094
+        IpProtocol: tcp
+        SourceSecurityGroupId: !GetAtt KafkaClientInstanceSecurityGroup.GroupId
+      - Description: Kafka Connect
+        FromPort: 8083
+        ToPort: 8083
+        IpProtocol: tcp
+        SourceSecurityGroupId: !GetAtt KafkaClientInstanceSecurityGroup.GroupId
+      - Description: Enable access to Schema Registry inside the SG
+        FromPort: 8081
+        ToPort: 8081
+        IpProtocol: tcp
+        SourceSecurityGroupId: !GetAtt KafkaClientInstanceSecurityGroup.GroupId
+
+  MSKSecurityGroup9094:
+    Type: AWS::EC2::SecurityGroupIngress
+    DependsOn: MSKSecurityGroup
+    Properties:
+      GroupId: !GetAtt MSKSecurityGroup.GroupId
+      Description: Enable Self referencing Bootstrap servers TLS
+      IpProtocol: tcp
+      FromPort: 9094
+      ToPort: 9094
+      SourceSecurityGroupId: !GetAtt MSKSecurityGroup.GroupId
+
+  MSKSecurityGroup9092:
+    Type: AWS::EC2::SecurityGroupIngress
+    DependsOn: MSKSecurityGroup
+    Properties:
+      GroupId: !GetAtt MSKSecurityGroup.GroupId
+      Description: Enable Self referencing Bootstrap servers Plaintext
+      IpProtocol: tcp
+      FromPort: 9092
+      ToPort: 9092
+      SourceSecurityGroupId: !GetAtt MSKSecurityGroup.GroupId
+
+  MSKSecurityGroup9096:
+    Type: AWS::EC2::SecurityGroupIngress
+    DependsOn: MSKSecurityGroup
+    Properties:
+      GroupId: !GetAtt MSKSecurityGroup.GroupId
+      Description: Enable Self referencing SASL
+      IpProtocol: tcp
+      FromPort: 9096
+      ToPort: 9096
+      SourceSecurityGroupId: !GetAtt MSKSecurityGroup.GroupId
+
+  MSKSecurityGroup9098:
+    Type: AWS::EC2::SecurityGroupIngress
+    DependsOn: MSKSecurityGroup
+    Properties:
+      GroupId: !GetAtt MSKSecurityGroup.GroupId
+      Description: Enable Self referencing IAM
+      IpProtocol: tcp
+      FromPort: 9098
+      ToPort: 9098
+      SourceSecurityGroupId: !GetAtt MSKSecurityGroup.GroupId
+
+  MSKCluster:
+    Type: AWS::MSK::Cluster
+    Properties: 
+      BrokerNodeGroupInfo: 
+        ClientSubnets:
+          - !Ref PrivateSubnetMSKOne
+          - !Ref PrivateSubnetMSKTwo
+        SecurityGroups:
+          - !GetAtt MSKSecurityGroup.GroupId
+        InstanceType: !Ref InstanceType
+        StorageInfo:
+          EBSStorageInfo:
+            VolumeSize: 500
+      ClusterName: !Sub "${AWS::StackName}-cluster"
+      ClientAuthentication:
+          Sasl:
+            Iam:
+              Enabled: True
+      KafkaVersion: !Ref MSKKafkaVersion
+      NumberOfBrokerNodes: 2
+      EncryptionInfo: 
+        EncryptionInTransit: 
+          InCluster: true
+
+  Cloud9EC2Bastion:
```

Comment:
This is failing to deploy: 
CREATE_FAILED                               AWS::CloudFormation::Stack                  MSKCluster                                  Embedded stack arn:aws:cloudformation:us- 
                                                                                                                                    east-1:978558897928:stack/eventbridge-    
                                                                                                                                    pipes-msk-iam-auth-to-lambda-MSKCluster-1 
                                                                                                                                    3J5OA575F6GI/8d313870-b9d1-11ed-          
                                                                                                                                    ad9a-0eac6f74b859 was not successfully    
                                                                                                                                    created: The following resource(s) failed 
                                                                                                                                    to create: [Cloud9EC2Bastion,             
                                                                                                                                    MSKCluster].

Reply:
What does this bastion host do? In your testing instructions you say deploy 2 x EC2 instances for testing. Can you use the bastion host for this?

Reply:
Added instructions to make this clearer. The template now provides an optional (commented code) Cloud 9 and EC2 instance deployed in private subnet which has all the Kafka producer components setup.